### PR TITLE
Migrate mysqlctld from flag to pflag

### PR DESCRIPTION
**Pull request by mattlord**
## Description

This moves the `mysqlctld` command/binary from using flag to pflag.

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/11267

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]
LGTM!

File: go/cmd/mysqlctld/mysqlctld.go [hunk 1]
```diff
@@ -54,14 +55,21 @@ func init() {
...
 	servenv.RegisterGRPCServerAuthFlags()
 	servenv.RegisterServiceMapFlag()
+	// mysqlctld only starts and stops mysql, only needs dba.
+	dbconfigs.RegisterFlags(dbconfigs.Dba)
+	servenv.OnParse(func(fs *pflag.FlagSet) {
+		fs.IntVar(&mysqlPort, "mysql_port", mysqlPort, "mysql port")
+		fs.UintVar(&tabletUID, "tablet_uid", tabletUID, "tablet uid")
+		fs.StringVar(&mysqlSocket, "mysql_socket", mysqlSocket, "path to the mysql socket")
+		fs.DurationVar(&waitTime, "wait_time", waitTime, "how long to wait for mysqld startup or shutdown")
+		fs.StringVar(&initDBSQLFile, "init_db_sql_file", initDBSQLFile, "path to .sql file to run after mysql_install_db")
```
Comment (GuptaManan100) [inline go/cmd/mysqlctld/mysqlctld.go:65]: Just one comment in hindsight. We should make the help for these commands start with capital case. This is not really related to flags change, but something we want to fix going forward

Comment (mattlord) [reply go/cmd/mysqlctld/mysqlctld.go:65]: Will do! I thought about it doing it at the time, but then left it as I wasn't sure we had all agreed on that (I have been doing this). 🙂

Comment (mattlord) [reply go/cmd/mysqlctld/mysqlctld.go]: Just FYI, since I was there I also cleaned up the text a bit (for example, `mysql_install_db` is no longer used for init): https://github.com/vitessio/vitess/pull/11376/commits/21f8c2306c06fb65759b6335f67c77899424ff4d
